hngx-org/zuriportfolio-commander-backend
# Feat : get all orders

# PR Title: Implements Linear Ticket

> **Implements Linear Ticket** #COM-22 

## Overview

This PR introduces two significant enhancements to our application's order retrieval functionality. These enhancements are detailed below:

- ### Improved Pagination
  - **Description**: This feature improves user experience by enabling the retrieval of orders in manageable chunks, preventing long response times when dealing with a large number of orders.
  - **User Benefit**: Users can now specify the page they want to view and set the page size according to their preference.

## Ticket Details

- **Ticket**: [COM-22](https://linear.app/zuri-project-backend/issue/COM-22/api-to-get-all-orders)

## Checklist

- [x] The code adheres to the project's coding style.
- [x] The PR title and description are clear, providing a comprehensive explanation of the approach taken.
- [x] The pull request is targeting the **main branch** (left side).
- [x] Commit messages conform to the requested structure.
- [x] Code additions do not fail code linting checks or unit tests.
- [x] Changes are limited to the requested files.

## Usage

### Testing Steps

1. Acquire a valid authentication token (without which you may encounter an unauthorized error).
2. Create a .env file with your DATABASE_URL variable.
3. Please ensure to Populate your database very well oooooo !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
4. Start the server and launch your test client (e.g., Postman).
5. Send a GET request to /api/orders.

## Screenshots/Videos

This image illustrates the response structure, providing detailed information about an order. It includes the order's creation timestamp, details about the associated merchant, customer, and the product linked to the order. The response confirms a successful order retrieval operation with a "Orders fetched successfully" status and no reported errors (errorStatus set to false).

<img width="711" alt="image" src="https://github.com/hngx-org/zuriportfolio-commander-backend/assets/143452040/d79d6a81-ad3d-45b8-8a18-0978bee63937">

The image below shows the rest of the response.

<img width="707" alt="image" src="https://github.com/hngx-org/zuriportfolio-commander-backend/assets/143452040/49824112-cd46-4530-9cf2-6f8829726b5a">

<img width="700" alt="image" src="https://github.com/hngx-org/zuriportfolio-commander-backend/assets/143452040/0b6c2562-e83b-4510-9015-930c82b7122d">

<img width="704" alt="image" src="https://github.com/hngx-org/zuriportfolio-commander-backend/assets/143452040/f4ef9e8d-edd5-4d10-ae4f-9cc1708006d8">





The image below shows the response for an unauthorized user coming from the authentication service:

![Unauthorized Response](https://github.com/hngx-org/zuriportfolio-commander-backend/assets/144495013/940f29ec-b60f-4fad-bd87-fc987777d1cc)
